### ISSUE TITLE
The offered filename for exporting a recovery key uses the provider name in the invite instead of the name I assigned

### ISSUE THREAD

**Issue by exarkun** [gridsync/gridsync#68]
I accepted a provider invite for a provider I already have configured.

Should this even have configured a new provider?  Maybe, I don't know, but it did.

When I went to export the recovery key, the default filename in the save dialog included the _original_ name of the provider instead of the de-collided name I had entered.

I think it should use the name I assigned since this is more likely to have meaning for me.  Also, I already had a file with the name it suggested since I had already exported a recovery key for the first configuration of that provider.

**Comment by crwood** [comment]
I was able to reproduce this and am going to consider it a bug, caused by oversight on my part..

Basically, originally, Gridsync _always_ configured a new provider when joining a grid (in order to facilitate, e.g., users who might want to toggle between multiple "identities" within the same grid), however, with the advent of magic-folder sharing, this functionality needed to go away (since, for one, joining a magic-folder requires that Gridsync be able to figure out to which grid a given folder belongs -- something which it currently does by comparing the introducer fURL provided in the JSON blob against the list of already-known introducers).. In short, configuring a new provider in the event of a name-conflict should no longer even be a thing: Gridsync should only configure a new provider if and only if it hasn't already been joined (as it currently does for magic-folder invites). In other words, if a user tries joining a grid twice, the second time should inform them that they've already joined that grid (rather than prompting for a rename and configuring a second, redundant client).

TL;DR: No, it should not configure a new provider. I'll fix this..
